pages/api/sitemap.ts

```diff
@@ -0,0 +1,71 @@
+import * as path from "path";
+import globby from "globby";
+
+import { NextApiRequest, NextApiResponse } from "next";
+import {
+  getAllPostsWithSlug,
+  getAllProductCategoriesWithSlug,
+  getAllProductsWithSlug,
+} from "lib";
+
+type EdgeSlug = {
+  edges: { node: { slug: string } }[];
+};
+
+const lambdaPagesFilePath = path.join(__dirname, ".", "lib", "sitemap.xml");
```

Comment:
This looks brilliant if it work, change `lib` to `public` at the end is that the right place to save this file, do you think we can call this endpoint at build time?